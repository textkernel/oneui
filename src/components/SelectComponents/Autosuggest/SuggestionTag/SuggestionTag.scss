.SuggestionTag {
    display: inline-flex;
    height: 24px;
    justify-content: space-between;
    align-items: center;
    font-size: var(--font-size-normal);
    padding: 0 var(--spacing-normal);
    background-color: var(--color-light-25);
    border-radius: var(--border-radius);
    margin: var(--spacing-normal);
    margin-top: var(--margin);
    margin-bottom: 4px;
    max-width: 200px;
    min-width: 15px;
    cursor: default;

    &--block {
        flex: 1 1 auto;
    }

    &--small {
        flex: 1 0 auto;
        max-width: 36px;
    }

    &__label {
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
        padding: 0 var(--spacing-2x) 0;
        color: var(--color-foreground);
    }

    &__button {
        border: 0;
        cursor: pointer;
        background-color: var(--transparent);
        padding: 0;

        svg {
            font-size: var(--font-size-large);
            vertical-align: middle;
        }
    }
}
