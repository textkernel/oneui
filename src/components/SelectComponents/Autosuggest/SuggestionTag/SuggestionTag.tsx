import * as React from 'react';
import { MdClose } from 'react-icons/md';
import { bem } from '../../../../utils';
import styles from './SuggestionTag.scss';

const { block, elem } = bem('SuggestionTag', styles);

interface Props {
    /** Defines with behavior for tag */
    width?: 'block' | 'small' | 'auto';
    /** Clicked on delete button */
    onClick?: (e: React.MouseEvent) => void;
    /** Tag content */
    children: NotEmptyReactNode;
}

export const SuggestionTag: React.FC<Props> = ({ width = 'auto', children, onClick, ...rest }) => {
    return (
        <div {...rest} {...block({ [width || 'auto']: true })}>
            <div title={children} {...elem('label')}>
                {children}
            </div>
            {onClick ? (
                <button onClick={onClick} type="button" {...elem('button')}>
                    <MdClose />
                </button>
            ) : null}
        </div>
    );
};

SuggestionTag.displayName = 'SuggestionTag';
