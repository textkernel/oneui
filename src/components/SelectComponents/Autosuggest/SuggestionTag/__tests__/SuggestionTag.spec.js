import React from 'react';
import toJson from 'enzyme-to-json';
import { SuggestionTag } from '../SuggestionTag';

describe('SuggestionTag', () => {
    it('should render nothing if no children are provided', () => {
        const wrapper = shallow(<SuggestionTag />);
        expect(wrapper.html()).toBeNull();
    });
    it('should render correctly', () => {
        const wrapper = mount(<SuggestionTag>tag</SuggestionTag>);

        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('button')).toHaveLength(0);
    });
    it('should render correctly with styles modifier', () => {
        const wrapper = mount(<SuggestionTag width="block">tag</SuggestionTag>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should add delete button if onClick is defined', () => {
        const onClickMock = jest.fn();
        const wrapper = mount(<SuggestionTag onClick={onClickMock}>tag</SuggestionTag>);

        expect(wrapper.find('button')).toHaveLength(1);
    });
    it('should call onClick callback correctly', () => {
        const onClickMock = jest.fn();
        const wrapper = mount(<SuggestionTag onClick={onClickMock}>tag</SuggestionTag>);

        wrapper.find('button').simulate('click');

        expect(onClickMock).toHaveBeenCalledTimes(1);
    });
});
