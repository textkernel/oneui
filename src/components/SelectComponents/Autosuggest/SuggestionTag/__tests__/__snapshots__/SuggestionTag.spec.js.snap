// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestionTag should render correctly 1`] = `
<SuggestionTag>
  <div
    className="SuggestionTag"
  >
    <div
      className="SuggestionTag__label"
      title="tag"
    >
      tag
    </div>
  </div>
</SuggestionTag>
`;

exports[`SuggestionTag should render correctly with styles modifier 1`] = `
<SuggestionTag
  width="block"
>
  <div
    className="SuggestionTag SuggestionTag--block"
  >
    <div
      className="SuggestionTag__label"
      title="tag"
    >
      tag
    </div>
  </div>
</SuggestionTag>
`;
